src/java/picard/sam/markduplicates/util/OpticalDuplicateFinder.java

```diff
@@ -67,45 +67,36 @@ public OpticalDuplicateFinder(final String readNameRegex, final int opticalDupli
     /**
      * Finds which reads within the list of duplicates are likely to be optical duplicates of
      * one another.
-     * <p/>
-     * Note: this method will perform a sort() of the list; if it is imperative that the list be
-     * unmodified a copy of the list should be passed to this method.
      *
      * @param list a list of reads that are determined to be duplicates of one another
+     * @param keeper a single PhysicalLocation that is the one being kept as non-duplicate, and thus should never be
+     *               annotated as an optical duplicate. May in some cases be null, or a PhysicalLocation not
+     *               contained within the list!
      * @return a boolean[] of the same length as the incoming list marking which reads are optical duplicates
      */
-    public boolean[] findOpticalDuplicates(final List<? extends PhysicalLocation> list) {
+    public boolean[] findOpticalDuplicates(final List<? extends PhysicalLocation> list, final PhysicalLocation keeper) {
         final int length = list.size();
         final boolean[] opticalDuplicateFlags = new boolean[length];
+        final int distance = this.opticalDuplicatePixelDistance;
 
-        Collections.sort(list, new Comparator<PhysicalLocation>() {
-            public int compare(final PhysicalLocation lhs, final PhysicalLocation rhs) {
-                int retval = lhs.getReadGroup() - rhs.getReadGroup();
-                if (retval == 0) retval = lhs.getTile() - rhs.getTile();
-                if (retval == 0) retval = lhs.getX() - rhs.getX();
-                if (retval == 0) retval = lhs.getY() - rhs.getY();
-                return retval;
-            }
-        });
-
-        outer:
-        for (int i = 0; i < length; ++i) {
+        for (int i=0; i<length; ++i) {
```

Comment:
@yfarjoun , @eitanbanks , @nh13 : this method is now *much* simpler, though likely less efficient for large piles of duplicates.  None of us (me included) caught the fact that the method returns an array of flags that is index-synced to the list of PhysicalLocations - but only the post-sorted version.  That seems really nasty and error-prone, so I removed the sort and simplified the code.